File: sentimental-mario-more/mario.py
from cs50 import get_int


def main():

    height = get_int("What's the height? ")

    while (not (height > 0 and height < 9)):

        height = get_int("What's the height? ")

    if (height > 0 and height < 9):
        spaces = height - 1
        for i in range(height):
            gap(spaces)
            dashes(i + 1)
            print("  ", end="")
            dashes(i + 1)
            print()
            spaces = spaces - 1


def gap(spaces):
    for i in range(spaces):
        print(" ", end="")


def dashes(length):
    for i in range(length):
        print("#", end="")


main()

File: mario.c
#include <cs50.h>
#include <stdio.h>

void gap(int spaces);
void dashes(int length);

int main(void)
{
    int height = get_int("What's the height? ");

    while (!(height > 0 && height < 9))
    {
        height = get_int("What's the height? ");
    }

    if (height > 0 && height < 9)
    {
        int spaces = height - 1;
        for (int i = 0; i < height; i++)
        {
            gap(spaces);
            dashes(i + 1);
            printf("  ");
            dashes(i + 1);
            printf("\n");
            spaces--;
        }
    }
}

void gap(int spaces)
{
    for (int i = 0; i < spaces; i++)
    {
        printf(" ");
    }
}

void dashes(int length)
{
    for (int i = 0; i < length; i++)
    {
        printf("#");
    }
}
